lib/kubeclient/common.rb

```diff
@@ -207,34 +207,31 @@ def delete_entity(entity_type, name, namespace = nil)
     end
 
     def create_entity(entity_type, entity_config, klass)
-      # to_hash should be called because of issue #9 in recursive open
-      # struct
+      # Duplicate the entity_config to a hash so that when we assign
+      # kind and apiVersion, this does not mutate original entity_config obj.
       hash = entity_config.to_hash
 
-      ns_prefix = build_namespace_prefix(entity_config.metadata['table'][:namespace])
+      ns_prefix = build_namespace_prefix(hash[:metadata][:namespace])
 
       # TODO: temporary solution to add "kind" and apiVersion to request
       # until this issue is solved
       # https://github.com/GoogleCloudPlatform/kubernetes/issues/6439
-      hash['kind'] = entity_type
-      hash['apiVersion'] = @api_version
+      hash[:kind]       = entity_type
+      hash[:apiVersion] = @api_version
       response = handle_exception do
         rest_client[ns_prefix + resource_name(entity_type)]
-        .post(hash.to_json, @headers)
+        .post(entity_config.to_json, @headers)
```

Comment:
shouldn't it be hash that is sent to the server? 
the entity_config doesn't contain kind and apiVersion set on lines 219 and 220